uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences.js

```diff
@@ -0,0 +1,32 @@
+var Promise = require('es6-promise').Promise;
+var page = require('./page');
+
+module.exports = (function () {
+
+  return page.extend({
+    /**
+     * [init page]
+     */
+    initPage: function () {
+      var casper = this.casper;
+
+      return new Promise(function (resolve) {
```

Comment:
I think what you're doing here is another implementation of what is already happening in the main page's object
```js
/**
 * Initializes the page
 *
 * Stores a customized version of CasperJS and then wait for a
 * until a certain "ready" condition is met, if the page is set up to do so
 */
init: function (casper, clearDialogs) {
  !!this.waitForReady && this.casper.then(function () {
    this.waitForReady();
  }.bind(this));
}
```
if that's the case, then this is not necessary. Please check what the current architecture offers before adding new methods